dev-docs/plugins/bc/bc-prebid-plugin-building-adapter.md

```diff
@@ -0,0 +1,205 @@
+
+Using an adapter is particularly helpful when the plugin is defined and configured in Brightcove Studio.  While configuring the plugin in the Studio is very convenient and removes the need to define the plugin on the publisher's page, it limits the ability to make decisions about the prebid process at run-time.  Building adapters is a way to add custom behavior to the plugin based on run-time conditions.
+
+## How Adapters Work
+
+A publisher defines custom behavior using an external Javascript file.  The plugin is informed about the existence of the adapter file in the plugin configuration. The plugin loads the specified adapter(s) using the URL cited in the adapter definition (see below).
+
+Once the adapter script has been loaded, the adapter will attach itself to the window object of the document where the plugin has been loaded.  The adapter uses its unique identifier (the same one that is also specified in the Prebid configuration) to store a reference to itself on the window object. The plugin uses this reference to query the adapter for status.  As an example, if the identifier (`id`) in the configuration is defined as `"myPluginAdapter"`, the reference to the loaded adapter can be retrieved from `window.myPluginAdapter` in the same document where the plugin has been loaded.
+
+At certain points during the running of the plugin, points that correspond to the known API entry points of adapters, the plugin will query the adapter for status using one of the API entry points supported by the adapter. Depending on the outcome of the query, the Prebid plugin may or may not change its default behavior.
+
+## How to Define an Adapter to the Plugin
+
+Adapters are defined in the `adapters` prebid configuration option.  This is an optional configuration setting.  If you do want to define one or more adapters, you need to set the value of the `adapters` option to an array of one or more adapter definitions, as shown below.
+
+### Adapter Definition
+
+Each adapter definition is an object, which should contain the following fields:
+
+- `id` - this is the unique identifier of the adapter.
+    - This value should be a string; it should be the same value that the adapter is going to use to store the references to itself after it has been loaded.
+    - This is also the same identifier that the Prebid plugin is going to use to communicate with the adapter once it has been loaded.
+- `url` - this is the URL that the plugin will use to load in the adapter script.
+
+### Example Adapter Configuration
+
+```
+ options.adapters = [{id : 'myPluginAdapter', url : 'https://my-path/my-plugin-adapter.js'}];
+```
+
+### More Information
+
+For more information about adapter configuration, see  **[Prebid Options]({{site.baseurl}}/dev-docs/plugins/bc/bc-prebid-plugin-prebid-options.html#pb-adapters)**.
+
+## Adapter API
+
+By default, the Brightcove Prebid Plugin supports the following adapter API method:
```

Comment:
let's change this to API methods (even though we have only one entry method right now).  This will make it easier to add additional entry methods later.

Reply:
I'd really like to leave this as is. If we change it to plural but only list one, it gives the impression something is missing.

Reply:
Sure ... that is fine.